### PR TITLE
Syntax: Treat typespecs as an documentation

### PR DESCRIPTION
**Pull request by ayarulin**
Treating typespecs as an documentation, makes code looks cleaner and keeps focus on actual method implementation.

### CODE REVIEW

**Review by joaothallis — APPROVED** [review]

**Comment by jeroenvisser101** [comment]
@ayarulin this would also disable any syntax highlighting for `@type` definitions, which sometimes can be quite helpful. This is also true in the other definitions but especially with type definitions.
image

File: syntaxes/elixir.json [hunk 1]
```diff
@@ -19,6 +19,12 @@
       "match": "([A-Z]\\w*\\s*(\\.)\\s*)*[A-Z]\\w*",
       "name": "entity.name.type.module.elixir"
     },
+    {
+      "begin": "@(spec|type|typep)\\s",
```
Comment (emadb) [inline syntaxes/elixir.json]: @jeroenvisser101 is right. 
Probably here is better to match only the `@spec` annotation:

```
 "begin": "@spec\\s",
```

**Comment by jeroenvisser101** [comment]
Could we at all make this configurable?

**Comment by joecorkerton** [comment]
Would be great if this can be added as a config option. I find the current highlighting a bit distracting as it has the same 'prominence' as the rest of the code. I think having a config to also do the same for `@type` annotations would be great too.

**Comment by JakeBecker** [comment]
This project has moved!

It's now being maintained by proactive volunteers from the Elixir community over at [elixir-lsp](https://github.com/elixir-lsp)/[vscode-elixir-ls](https://github.com/elixir-lsp/vscode-elixir-ls). The main ElixirLS codebase is now at [elixir-lsp](https://github.com/elixir-lsp)/[elixir-ls](https://github.com/elixir-lsp/elixir-ls). Updates will continue to be published from that repo to the original VS Code extension, so no need to switch plugins.

If you're still interested in merging this PR, please do the following:

1. Check that the issue you're addressing still occurs with the latest version of the VS Code plugin (which is published from the new repo)
2. If your PR references a Github issue, check if there's already an issue filed on the new repo and create one if not
3. Rebase your PR from the new repo (which has diverged significantly) and re-submit it there.

Thank you!

**Pull request closed without merging**